napari/components/dims.py

```diff
@@ -325,15 +380,50 @@ def set_mode(self, axis: int, mode: DimsMode):
         mode : POINT or INTERVAL
             Whether dimension is in the POINT or INTERVAL mode
         """
+        axis = self._assert_axis_inbound(axis)
+        if self.mode[axis] != mode:
+            self._mode[axis] = mode
+            self.events.axis(axis=axis)
+
+    def set_axis_label(self, axis: int, label: str):
+        """Sets a new axis label for the given axis.
+
+        Parameters
+        ----------
+        axis : int
+            Dimension index
+        label : str
+            Given label
+        """
+        axis = self._assert_axis_inbound(axis)
+        if self.axis_labels[axis] != str(label):
+            self._axis_labels[axis] = str(label)
+            self.events.axis_labels(axis=axis)
+
+    def _assert_axis_inbound(self, axis: int) -> int:
+        """Asserts that a given value of axis is inside
+        the existing axes of the image.
+
+        Returns
+        -------
+        axis : int
+            The axis which was checked for validity
+
+        Raises
+        ------
+        ValueError
+            The given axis index is out of bounds
+        """
         if axis < 0:
             axis += self.ndim
         if axis < 0:
+            raise ValueError(f'Axis is too negative, got {axis}.')
```

Comment:
Heh. "too negative" is funny but imho doesn't quite capture it. And anyway the axis value printed is not what the function got. I suggest:

```python
if axis not in range(-self.ndim, self.ndim):
    msg = (f'Axis {axis} not defined for dimensionality {self.ndim}. '
           f'Must be in [{-self.ndim}, {self.ndim}).')
    raise ValueError(msg)

return axis % self.ndim
```

I also think it should be `in_bounds` instead of `inbound`, otherwise I think of "assert axis is incoming" 😄